src/main/java/com/google/devtools/build/lib/analysis/platform/ConstraintValueInfo.java

```diff
@@ -73,13 +74,19 @@ public Label label() {
    * method.
    */
   public ConfigMatchingProvider configMatchingProvider(PlatformInfo platformInfo) {
+    ConstraintValueInfo platformValue = platformInfo.constraints().get(this.constraint());
     return ConfigMatchingProvider.create(
         label,
         ImmutableMultimap.of(),
         ImmutableMap.of(),
         ImmutableSet.of(),
-        ConfigMatchingProvider.MatchResult.create(
-            platformInfo.constraints().hasConstraintValue(this)));
+        equals(platformValue)
```

Comment:
Change to `this.equals(platformValue)`. because I was very confused trying to figure out what was happening here.

It might also be nice to combine the entire thing into a new `computeMatchResult` that does the `equals` check and creates the `MATCH` or `NoMatch` instance.